### PR TITLE
Deprecate security_posture_config in google_container_attached_cluster

### PR DESCRIPTION
**Pull request by hdp617**
and add deprecation message.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:deprecation
containerattached: deprecated `security_posture_config` field in `google_container_attached_cluster` resource
```

### CODE REVIEW

File: mmv1/products/containerattached/Cluster.yaml [hunk 1]
```diff
@@ -360,6 +360,7 @@ properties:
     description: |
       Enable/Disable Security Posture API features for the cluster.
     default_from_api: true
+    deprecation_message: '`security_posture_config` is deprecated and will be removed in a future release.'
```
Comment (zli82016) [inline mmv1/products/containerattached/Cluster.yaml:363]: Will this field be replaced by a new field or just be removed?

Comment (hdp617) [reply mmv1/products/containerattached/Cluster.yaml:363]: It will be removed.

Comment (zli82016) [inline mmv1/products/containerattached/Cluster.yaml:363]:
```suggestion
    deprecation_message: '`security_posture_config` is deprecated and will be removed in a future major release.'
```

**Review by zli82016 — APPROVED** [review]
LGTM

**Comment by wyardley** [comment]
Is / was this related to #19751?

**Comment by zli82016** [comment]
> Is / was this related to #19751?

Yes, it is part of https://github.com/hashicorp/terraform-provider-google/issues/19751

**Comment by zli82016** [comment]
The service team decided not to launch this field, so this field is deprecated.

**Pull request merged**